torch/csrc/distributed/c10d/ProcessGroupGloo.cpp

```diff
@@ -1919,6 +1919,38 @@ class AsyncAllgatherCUDAWork : public AsyncAllgatherWork {
 
 } // namespace
 
+c10::intrusive_ptr<Work> ProcessGroupGloo::_reduce_scatter_base(
```

Comment:
Nice, maybe worth removing those fallbacks in the functional collective paths and this might help testing https://github.com/pytorch/pytorch/blob/main/torch/distributed/_functional_collectives_impl.py#L180